apis/core/v1alpha2/featuregate_types.go

```diff
@@ -75,6 +76,7 @@ type FeatureGate struct {
 	Spec FeatureGateSpec `json:"spec,omitempty"`
 	// Status reports activation state and availability of features in the system.
 	Status FeatureGateStatus `json:"status,omitempty"`
+	config *rest.Config      `json:"-"`
```

Comment:
We should not have any API changes to support tests.

Reply:
It seems not possible to have envtest without this change. 

When we create a feature gate resource from envtest, [ValidateCreate](https://github.com/vmware-tanzu/tanzu-framework/blob/99dd6215d05ae98aa5b21f0553f4223a2c25e608/apis/core/v1alpha2/featuregate_webhook.go#LL69C23-L69C37) gets called. From `ValidateCreate`, [getClient](https://github.com/vmware-tanzu/tanzu-framework/blob/99dd6215d05ae98aa5b21f0553f4223a2c25e608/apis/core/v1alpha2/featuregate_webhook.go#L40) gets called. In `getClient`, the [in-cluster config](https://github.com/vmware-tanzu/tanzu-framework/blob/99dd6215d05ae98aa5b21f0553f4223a2c25e608/apis/core/v1alpha2/featuregate_webhook.go#LL50C2-L50C2) is obtained for the client. 

So, when we run envtest without this change, the webhook gets the client using in-cluster config and tries to query FeatureGate resource. That errors out. So, this change seems to be necessary. Please let me know if there are alternate approaches. 

Also, this change does not export any new fields. So, this should be fine.

Reply:
I think it is a valid point. I looked at other webhooks. They maintain two different structs for webhook and the API types so that any data required for webhook does not bring in API change. Let me have another look at this problem.

Reply:
Reverted API changes.